### ISSUE TITLE
warnings when compiling elektra

### ISSUE THREAD

**Issue by haraldg** [ElektraInitiative/libelektra#915]
I just recompiled elektra for OpenWRT (32bit arm target) and saved a log of the warnings. Maybe somebody wants to look at them and clean things up a bit:
http://friends.ccbib.org/lambda/elektra.warn

**Comment by markus2330** [comment]
Thank you for reporting! Seems like the warnings come from generated code thus there are so many of them. Most are related to wrong printf flags, so it should be fixable by using kdbtypes.h.

**Comment by tom-wa** [comment]
```
  /home/lambda/olinuxino/openwrt/build_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/elektra-0.8.17/src/plugins/resolver/filename.c:79:43: warning: format '%d' expects argument of type 'int', but argument 6 has ty    pe 'suseconds_t {aka long int}' [-Wformat=]
    snprintf (where + len, POSTFIX_SIZE - 1, ".%d:%ld." ELEKTRA_TIME_USEC_F ".tmp", getpid (), tv.tv_sec, tv.tv_usec);
                                             ^
```
seems to be a false positive / preprocessor problem

not sure what todo about this one:
```
/home/lambda/olinuxino/openwrt/staging_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/usr/include/boost/type_traits/detail/bool_trait_def.hpp:18:79: note: #pragma message: NOTE: Use of this header (bool_trait_def    .hpp) is deprecated
 # pragma message("NOTE: Use of this header (bool_trait_def.hpp) is deprecated")
```

**Comment by markus2330** [comment]
> seems to be a false positive / preprocessor problem

Seems like that kdbtypes.h needs an additional `#ifdef` here.

> not sure what todo about this one

We should tag the include path of boost with CMake Flag SYSTEM (will use gcc flag `-isystem`). This should avoid such warnings then.

**Comment by tom-wa** [comment]
>We should tag the include path of boost with CMake Flag SYSTEM (will use gcc flag -isystem). 

not sure what you mean.

**Comment by markus2330** [comment]
https://cmake.org/cmake/help/v3.0/command/include_directories.html

so you do something like include_directories(SYSTEM ${BOOST_INCL})

Where does the error occur? (Which plugin?)

**Comment by tom-wa** [comment]
>Where does the error occur? (Which plugin?)

tlc

**Comment by markus2330** [comment]
Ok, I fixed it in 97654f5e293f5c0de61041f3febc3ad5ebdbb1a6

Can you please reduce #946 to only fix the second part with kdbtypes.h?
